src/backend/vector/neon/field.rs

```diff
@@ -0,0 +1,757 @@
+        use core::arch::aarch64::vset_lane_u32;
+        use core::arch::aarch64::vgetq_lane_u32;
+
+        (vcombine_u32(
+                vset_lane_u32(vgetq_lane_u32(x.0.into_bits(), 2) , vget_low_u32(x.0.into_bits()), 1),
+                vset_lane_u32(vgetq_lane_u32(y.0.into_bits(), 2) , vget_low_u32(y.0.into_bits()), 1)).into_bits(),
+         vcombine_u32(
+                vset_lane_u32(vgetq_lane_u32(x.1.into_bits(), 2) , vget_low_u32(x.1.into_bits()), 1),
+                vset_lane_u32(vgetq_lane_u32(y.1.into_bits(), 2) , vget_low_u32(y.1.into_bits()), 1)).into_bits())
+    }
+}
+
+#[derive(Copy, Clone, Debug)]
+pub enum Lanes {
+    C,
+    D,
+    AB,
+    AC,
+    CD,
+    AD,
+    BC,
+    ABCD,
+}
+
+#[derive(Copy, Clone, Debug)]
+pub enum Shuffle {
+    AAAA,
+    BBBB,
+    CACA,
+    DBBD,
+    ADDA,
+    CBCB,
+    ABAB,
+    BADC,
+    BACD,
+    ABDC,
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct FieldElement2625x4(pub(crate) [(u32x4, u32x4); 5]);
+
+use subtle::Choice;
+use subtle::ConditionallySelectable;
+
+impl ConditionallySelectable for FieldElement2625x4 {
+    fn conditional_select(
+        a: &FieldElement2625x4,
+        b: &FieldElement2625x4,
+        choice: Choice,
+    ) -> FieldElement2625x4 {
+        let mask = (-(choice.unwrap_u8() as i32)) as u32;
+        let mask_vec = u32x4::splat(mask);
+        FieldElement2625x4([
+            (a.0[0].0 ^ (mask_vec & (a.0[0].0 ^ b.0[0].0)), a.0[0].1 ^ (mask_vec & (a.0[0].1 ^ b.0[0].1))),
+            (a.0[1].0 ^ (mask_vec & (a.0[1].0 ^ b.0[1].0)), a.0[1].1 ^ (mask_vec & (a.0[1].1 ^ b.0[1].1))),
+            (a.0[2].0 ^ (mask_vec & (a.0[2].0 ^ b.0[2].0)), a.0[2].1 ^ (mask_vec & (a.0[2].1 ^ b.0[2].1))),
+            (a.0[3].0 ^ (mask_vec & (a.0[3].0 ^ b.0[3].0)), a.0[3].1 ^ (mask_vec & (a.0[3].1 ^ b.0[3].1))),
+            (a.0[4].0 ^ (mask_vec & (a.0[4].0 ^ b.0[4].0)), a.0[4].1 ^ (mask_vec & (a.0[4].1 ^ b.0[4].1)))
+        ])
+    }
+
+    fn conditional_assign(
+        &mut self,
+        other: &FieldElement2625x4,
+        choice: Choice,
+    ) {
+        let mask = (-(choice.unwrap_u8() as i32)) as u32;
+        let mask_vec = u32x4::splat(mask);
+        self.0[0].0 ^= mask_vec & (self.0[0].0 ^ other.0[0].0);
+        self.0[0].1 ^= mask_vec & (self.0[0].1 ^ other.0[0].1);
+        self.0[1].0 ^= mask_vec & (self.0[1].0 ^ other.0[1].0);
+        self.0[1].1 ^= mask_vec & (self.0[1].1 ^ other.0[1].1);
+        self.0[2].0 ^= mask_vec & (self.0[2].0 ^ other.0[2].0);
+        self.0[2].1 ^= mask_vec & (self.0[2].1 ^ other.0[2].1);
+        self.0[3].0 ^= mask_vec & (self.0[3].0 ^ other.0[3].0);
+        self.0[3].1 ^= mask_vec & (self.0[3].1 ^ other.0[3].1);
+        self.0[4].0 ^= mask_vec & (self.0[4].0 ^ other.0[4].0);
+        self.0[4].1 ^= mask_vec & (self.0[4].1 ^ other.0[4].1);
+    }
+}
+
+impl FieldElement2625x4 {
+
+    pub fn split(&self) -> [FieldElement51; 4] {
+        let mut out = [FieldElement51::zero(); 4];
```

Comment:
```suggestion
        let mut out = [FieldElement51::ZERO; 4];
```

This was changed to ZERO
- https://github.com/dalek-cryptography/curve25519-dalek/pull/470